Source/WebCore/page/ContextMenuContext.h

```diff
@@ -82,7 +81,7 @@ class ContextMenuContext {
 private:
     Type m_type { Type::ContextMenu };
     HitTestResult m_hitTestResult;
-    RefPtr<Event> m_event;
+    WeakPtr<Event> m_event;
```

Comment:
I can't explain yet why this needs to be a WeakPtr, but I stop seeing leaked documents when running `notifyutil -p org.WebKit.lowMemory && notifyutil -p com.apple.WebKit.showAllDocuments` only when this is made WeakPtr.

Reply:
I'm guessing it's because the Event holds a RefPtr to an EventTarget. Nodes are EventTargets.

Reply:
You probably want to figure out the correct ownership semantics here. Making everything a weak pointer would probably solve all ref cycles but it would also be incorrect lifetime management.